soh/soh/Enhancements/mods.cpp

```diff
@@ -72,6 +74,31 @@ void RegisterAutoSaveOnReceiveItemHook() {
     });
 }
 
+void RegisterRupeeDash() {
+    GameInteractor::Instance->RegisterGameHook<GameInteractor::OnPlayerUpdate>([]() {
+        if (!CVarGetInteger("gRupeeDash", 0)) {
+            return;
+        }
+        
+        // Initialize Timer
+        static uint16_t rupeeDashTimer = 0;
+        uint16_t rdmTime = CVarGetInteger("gDashInterval", 5) * 20;
+        
+        // Did time change by DashInterval?
+        if (rupeeDashTimer >= rdmTime) {
+            rupeeDashTimer = 0;
+            if (gSaveContext.rupees > 0) {
+                Rupees_ChangeBy(-1);
+            } else {
+                Health_ChangeBy(gPlayState, -16);
```

Comment:
@dcvz should we use `GameInteractor::RawAction` here instead?

Reply:
my guess is no because we want to actually be able to go down to 0 health with it

Reply:
i looked into that and it seems like there's some extra logic in there with invincibility that i'm not sure we want to have for this